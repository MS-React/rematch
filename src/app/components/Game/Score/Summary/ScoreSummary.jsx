import React from 'react';
import PropTypes from 'prop-types';

const ScoreSummary = ({ results }) => (
  <div key="game-score-user-summay" className="score-summary">
    Total Score: {results.score} <br />
    Success: {results.success} <br />
    Failures: {results.fails}
    <br />
  </div>
);

ScoreSummary.propTypes = {
  results: PropTypes.shape({
    score: PropTypes.number,
    success: PropTypes.number,
    fails: PropTypes.number,
  }).isRequired,
};

export default ScoreSummary;
