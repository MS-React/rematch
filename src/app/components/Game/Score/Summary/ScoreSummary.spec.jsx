import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import ScoreSummary from './ScoreSummary';

describe('ScoreSummary', () => {
  let wrapper;
  let defaultProps;

  beforeEach(() => {
    defaultProps = {
      results: {
        success: 0,
        fails: 0,
        score: 0,
      },
    };
    wrapper = shallow(<ScoreSummary {...defaultProps} />);
  });

  it('should show user the summary reportto the user', () => {
    expect(renderer.create(wrapper).toJSON())
      .toMatchSnapshot();
  });
});
