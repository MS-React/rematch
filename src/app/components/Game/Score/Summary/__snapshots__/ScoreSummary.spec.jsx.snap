// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScoreSummary should show user the summary reportto the user 1`] = `
<div
  className="score-summary"
>
  Total Score: 
  0
   
  <br />
  Success: 
  0
   
  <br />
  Failures: 
  0
  <br />
</div>
`;
